CareKit/CareKit/Shared/Task/Controller/OCKTaskController.swift

```diff
@@ -444,6 +444,7 @@ open class OCKTaskController: ObservableObject {
         let detailViewController = OCKDetailViewController(showsCloseButton: true)
         detailViewController.detailView.titleLabel.text = task.title
         detailViewController.detailView.bodyLabel.text = task.instructions
+        detailViewController.detailView.imageView.image = task.asset.map(UIImage.init(named:)) ?? nil
```

Comment:
Looking at this again, I think we need to do something just a bit more nuanced. It's not guaranteed that the asset string is the name of an image in the assets catalog. It could also be an SFSymbol or name of a file in the app bundle.

There is a static method on `OCKContactUtility` that might be useful here.

```swift
    static func image(from asset: String?) -> UIImage? {
        guard let asset = asset else { return nil }

        // We can't be sure if the image they provide is in the assets folder, in the bundle, or in a directory.
        // We can check all 3 possibilities and then choose whichever is non-nil.
        let symbol = UIImage(systemName: asset)
        let appAssetsImage = UIImage(named: asset)
        let otherUrlImage = UIImage(contentsOfFile: asset)
        return otherUrlImage ?? appAssetsImage ?? symbol
    }
```

Reply:
Sounds good to me. Let me see if I can add it

Reply:
That uses optional binding for asset, should I go back to it?

Reply:
The reason that method takes an optional is so that you can write things like `image(from: task.asset)` cleanly on one line!